### PR TITLE
Feature/collection

### PR DESCRIPTION
**Pull request by ekate**

### CODE REVIEW

**Comment by ekate** [comment]
@jgpawletko 1 on 3 jobs passed. There are some Travis errors but i think they are related to problem with connection to fedora. For build to pass the collection object should be created in fedora, so when it is not we get the error

**Comment by jgpawletko** [comment]
Hi Kate,
Could you please take a look at the Jenkins job output?
The job failed to build, and there are deprecation warnings: 
```
DEPRECATION WARNING: You attempted to set the attribute `discoverable' on `Collection' to an enumerable value. However, this attribute is declared as being singular. This behavior is deprecated and will raise an ArgumentError in active-fedora 8.0.0. (called from block (2 levels) in object at /apps/wsops/.rbenv/versions/2.1.3/lib/ruby/gems/2.1.0/gems/factory_girl-4.4.0/lib/factory_girl/attribute_assigner.rb:16) 
```

And errors:
```
1) CollectionsController GET index should assign collection records to the collection instance variable
     Failure/Error: create(:collection)
     Rubydora::FedoraInvalidRequest:
       See logger for details
     # ./app/models/collection.rb:25:in `assign_pid'
     # ./spec/controllers/collections_controller_spec.rb:13:in `block (3 levels) in <top (required)>'
```

**Comment by ekate** [comment]
Hi Joe,

so you think the errors are not related to fedora connection ? Will keep looking

**Comment by jgpawletko** [comment]
@ekate 
Hi Kate,
I will look more closely too, but there were errors on Jenkins and Travis.

**Comment by ekate** [comment]
Joe,
About deprecation warning: I used the same tests for collection model as were used for NYUcore. One of the tests looks like that:
context 'when the value is incorrectly passed in as an Array' do
        let(:value) { ['1', '2'] }
        it "should get the #{field} attribute as a String" do
          expect(collection.send(field)).to be_present
          expect(collection.send(field)).to eq value.first
        end
      end
 
e.g. it on purpose sends data in wrong format hence the warning. We can get rid of those tests for both collection and nyucore. We also need to add validation for correct input (array/string) to avoid errors when system changes as warning says
Errors:
What happens: when collection object is created it requires pid. Noid service I added to the model generates random pid and then query fedora to check that the pid is unique. For whatever reasons fedora doesn't like that request from noid and marked it as wrong. I can't reproduce those errors locally and I don't know where to find more detailed logs (more then what I have in console output on jenkins) Do you know where I could look to see how the request was processed by Fedora ?
Thank you !

**Comment by jgpawletko** [comment]
Hi @ekate ,
Thank you for your analysis.
The hydra jetty logs should be written to `jetty/jettywrapper.log`.
I checked mine, however, and the output was pretty sparse.
I will look to see if there is a way to change the logging level, 
or if the log output is also written to another place.

**Comment by barnabyalter** [comment]
@ekate @jgpawletko The failure is happening on Travis, is it happening on Jenkins as well? If it is on Jenkins then you should be able to see whatever logs on there in the failing job. This inconsistency says to me that the tests are brittle and relying on the local environment too heavily. I can't tell quite what the invalid fedora request is but can you share where the code for `assign_pid` is in active_fedora? And where it falls in the callback chain?

Have you tried taking that method out of the equation and doing a process of assignment you know works, such as:

```
before_save :assign_pid

def assign_pid
  self.pid = "#{NOID_PREFIX}:#{noid_mint}"
end

def noid_mint
  @noid_mint ||= ActiveFedora::Noid::Service.new.mind
end
```

Because I can't find the source of that `assign_pid` function, its use is mysterious to me and it doesn't appear to be in any up-to-date codebases for the hydra stack so it may be something that is deprecated or removed.

On a related note, because of all the out of date gems and deprecations, the test output is very hard to parse with all the noise. I suggest updating dependencies before a deprecation turns into a removal. 

Also, I find it helpful when working with multiple coders with different styles to adopt the usage of `rubycritic` and `rubocop` to keep the code coder-agnostic.
